tests/test_sequence.py

```diff
@@ -300,6 +302,58 @@ def test_pred_alignment(self):
         self.assertEquals(preds, labels)
 
 
+class TestSequenceMemoryLeak(unittest.TestCase):
+
+    @staticmethod
+    def is_wr(val):
+        return val is not None and not isinstance(val, (int, float, str))
+
+    @staticmethod
+    def get_weakrefs(dictionary):
+        weakrefs = []
+        for v in dictionary.values():
+            if hasattr(v, "__dict__"):
+                weakrefs += TestSequenceMemoryLeak.get_weakrefs(v.__dict__)
+            if isinstance(v, dict):
+                weakrefs += TestSequenceMemoryLeak.get_weakrefs(v)
+            elif isinstance(v, list):
+                for vi in v:
```

Comment:
could make this check a bit broader (tuple, etc.)

Reply:
Feels like you could just recurse here rather than having the dict and type check... but might be missing something